tsconfig.json

```diff
@@ -1,8 +1,8 @@
 {
   "compilerOptions": {
     // Typechecking
-    "target": "ES2021",
-    "lib": ["ES2022", "DOM"],
+    "target": "ES2022",
+    "lib": ["ES2023"],
```

Comment:
This is a breaking change.

Reply:
I was under the impression that the ES bump isn't because that's what TypeScript [recommends](https://github.com/tsconfig/bases/blob/main/bases/node18.json) for Node 18 and is supported by [node.green](node.green). Turns out, they are [both](https://github.com/tsconfig/bases/issues/217) [wrong](https://github.com/compat-table/compat-table/issues/1874).

Removing the DOM lib should be safe though? It was added in #4851, most likely because `create-docusaurus` was run on the root instead of under the docusaurus workspace.